test/functions.js

```diff
@@ -390,22 +390,66 @@
     var counter = 0;
     var incr = function(){ counter++; };
     var throttledIncr = _.throttle(incr, 100);
-    var origNowFunc = _.now;
+    var originalNowFunc = Date.now;
+    var originalGetTimeFunc = Date.prototype.getTime;
 
     throttledIncr();
-    assert.strictEqual(counter, 1);
-    _.now = function() {
-      return new Date(2013, 0, 1, 1, 1, 1);
-    };
+    assert.strictEqual(counter, 1, '_.throttle: incr was called immediately');
+
+    Date.prototype.getTime = function() {
+      return +(new Date(2013, 0, 1, 1, 1, 1));
+    }
+    Date.now = function() {
+      return +(new Date(2013, 0, 1, 1, 1, 1));
+    }
 
     _.delay(function() {
       throttledIncr();
-      assert.strictEqual(counter, 2);
+      assert.strictEqual(counter, 2, '_.throttle: incr was called successfully, with tampered system time');
       done();
-      _.now = origNowFunc;
+      Date.now = originalNowFunc;
+      Date.prototype.getTime = originalGetTimeFunc;
     }, 200);
   });
 
+  QUnit.test('throttle continues to function after system time is not accessible (or in invalid format)', function(assert) {
+    assert.expect(3);
+    var done = assert.async();
+    var counter = 0;
+    var incr = function(){ counter++; };
+    var throttledIncr = _.throttle(incr, 100);
+    var originalNowFunc = Date.now;
+    var originalGetTimeFunc = Date.prototype.getTime;
+    var originalValueOfFunc = Date.prototype.valueOf;
+
+    throttledIncr();
+    assert.strictEqual(counter, 1, '_.throttle: incr was called immediately');
+
+    Date.prototype.valueOf = function() {
+      return null;
+    }
+    Date.prototype.getTime = function() {
+      return null;
+    }
+    Date.now = function() {
+      return null;
+    }
+
+    _.delay(function() {
+      throttledIncr();
+      assert.strictEqual(counter, 2, '_.throttle: incr was debounced successfully, with tampered system time');
```

Comment:
Throttling and debouncing are subtly different things, so to prevent confusion, I'd rather avoid the word "debounce" here. There is no convention to prefix all assertion comments with a qualified function name (at least not as far as I'm aware) and I don't think such a convention is necessary, either, so I suggest just changing it to this:

```suggestion
      assert.strictEqual(counter, 2, 'incr was throttled successfully, with tampered system time');
```